### ISSUE TITLE
Watch Test/Run Test in VSCode is sensitive to naming collision in nested describe blocks

### ISSUE THREAD

**Issue by eigilsagafos** [oven-sh/bun#15620]
### What version of Bun is running?

1.1.38+bf2f153f5

### What platform is your computer?

Darwin 24.1.0 arm64 arm

### What steps can reproduce the bug?

```ts
describe("group1", () => {
    describe("group2", () => {
        test("foo", () => {})
        test("bar", () => {})
    })
    test("foo", () => {})
    test("another test with foo in it", () => {})
    test("bar", () => {})
    test("another test with bar in it", () => {})
})
```

Result
```
✓ group1 > group2 > foo
» group1 > group2 > bar
✓ group1 > foo
✓ group1 > another test with foo in it
» group1 > bar
» group1 > another test with bar in it
```

### What is the expected behavior?

When hitting "Run Test/Watch Test" in VSCode on the `group1.group2.foo` is that just that test is run.

### What do you see instead?

Every test that matches `foo` will run

**Comment by RiskyMH** [comment]
I was wondering if this would be an issue, I believe it indeed uses the last string as the `--test-name-pattern` instead of them all separated with space.

**Comment by eigilsagafos** [comment]
> I was wondering if this would be an issue, I believe it indeed uses the last string as the `--test-name-pattern` instead of them all separated with space.

Makes sense. My current workaround is to add some random letters to the test I'm working on
